addon/modifiers/in-viewport.js

```diff
@@ -0,0 +1,114 @@
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { DEBUG } from '@glimmer/env';
+import Modifier from 'ember-class-based-modifier';
+import deepEqual from 'fast-deep-equal';
+
+const WATCHED_ELEMENTS = DEBUG ? new WeakSet() : undefined;
+
+export default class InViewportModifier extends Modifier {
+  @service inViewport;
+
+  name = 'in-viewport';
+
+  lastOptions;
```

Comment:
How does this work?  Just evaluates to `undefined`?

Reply:
Yes. Uninitialized class instance properties are `undefined`, but the property slot is already created. I add these out out of habit (TypeScript FTW), but supposedly this also improves performance because of [shaping](https://mathiasbynens.be/notes/shapes-ics). Not sure about that one though. The inner workings of JS engines are beyond my horizon. 🤷‍♂ 😄

Reply:
All for defining ALL THE SHAPES!